walrus_firmware/rosserial_teensyduino/teensyduino_sdk/arduino-1.0.6/hardware/teensy/cores/teensy/pins_teensy.c

```diff
@@ -1668,7 +1691,9 @@ static void disable_peripherals(void)
 
 void _reboot_Teensyduino_(void)
 {
-	cli();
+    wdt_enable(WDTO_250MS);
+    while(true);
+	/*cli();
```

Comment:
If this works why did you leave the other stuff commented out. If we don't need it delete it